src/unity/toolkits/activity_classification/activity_classifier.cpp

```diff
@@ -152,18 +154,25 @@ void activity_classifier::train(gl_sframe data, std::string target_column_name,
 {
   // Begin printing progress.
   // TODO: Make progress printing optional.
-  
-  // TODO: Support validation set.
-  training_table_printer_.reset(new table_printer(
+
+  if (variant_is<gl_sframe>(validation_data) ){
```

Comment:
It's looking like it could make sense to move the initialization of `training_table_printer_` into `init_train`, which is where we print the table header anyway.

(Also, let's make sure to run clang-format on this PR's commits too. We should aim for more consistent formatting than e.g. the parens spacing in this line.)